src/test/java/com/exasol/adapter/dialects/scalarfunction/VirtualSchemaTestTable.java

```diff
@@ -1,40 +1,16 @@
 package com.exasol.adapter.dialects.scalarfunction;
 
-import java.sql.SQLException;
-
-public class VirtualSchemaTestTable<T> {
-    private final String fullyQualifiedName;
-    private final SingleRowTableProvisioner<T> singleRowTableProvisioner;
+public interface VirtualSchemaTestTable {
 
     /**
-     * Create a new instance of {@link VirtualSchemaTestTable}.
-     *
-     * @param fullyQualifiedName        name of the table in the virtual schema
-     * @param singleRowTableProvisioner function that can change the value of the single row of the table
+     * Get the fully qualified name of the virtual schema table.
+     * 
+     * @return fully qualified name
      */
-    public VirtualSchemaTestTable(final String fullyQualifiedName,
-            final SingleRowTableProvisioner<T> singleRowTableProvisioner) {
-        this.fullyQualifiedName = fullyQualifiedName;
-        this.singleRowTableProvisioner = singleRowTableProvisioner;
-    }
-
-    public String getFullyQualifiedName() {
-        return this.fullyQualifiedName;
-    }
-
-    public void setValueOfSingleRow(final T value) throws SQLException {
-        this.singleRowTableProvisioner.initializedSingleRow(value);
-    }
+    public String getFullyQualifiedName();
 
-    @FunctionalInterface
-    public interface SingleRowTableProvisioner<T> {
-        /**
-         * Set the value of the single row of the table. Typical implementation: truncate table + insert row with the
-         * given value.
-         * 
-         * @param value value for the row
-         * @throws SQLException if something goes wrong
-         */
-        void initializedSingleRow(T value) throws SQLException;
-    }
+    /**
+     * Drop the virtual schema and the tables created for it.
+     */
+    public void drop();
```

Comment:
By looking at the comment of this method I realized this interface refers to a VirtualSchema rather to a virtual schema table.
I would suggest to rename it to something more meaningful, like `SingleTableVirtualSchema`, or simple `VirtualSchema`.